https://issues.redhat.com/browse/ACM-22631 Migration tech preview (#8270)
open-cluster-management/rhacm-docs: virtualization/migrate_vm.adoc

*Required access:* Cluster administrator

.Prerequsites

- You must have access to the `openshift-cnv` namespace.
- You have the `kubevirt.io:admin` role as both a target and a source cluster
- You have the `kubevirt.io-acm-hub:admin` for your hub cluster and `kubevirt.io-acm-managed:admin` for your target managed cluster. 

- You have the latest version of {ocp-short} installed. See the link:https://docs.redhat.com/en/documentation/openshift_container_platform/4.19/html/overview/index[{ocp-short} Overview].

To start migration for virtual machines complete the following steps:

. Enable `cnv-mtv-integrations-preview` component for the integration of {ocp-virt}. 
.. Run the following command to edit your `multiclusterhub` instance:

+
[source,bash]
----
oc edit MultiClusterHub multiclusterhub -n open-cluster-management
----

> .. From the _components_ section, set the `enabled` parameter to `true` for the `cnv-mtv-integrations-preview` component.

Author: @yiraeChristineKim confirmed that "components" is in the manifest so reverting back to using monospace: https://github.com/stolostron/rhacm-docs/pull/8270#discussion_r2466674509
```suggestion
.. From the `components` section, set the `enabled` parameter to `true` for the `cnv-mtv-integrations-preview` component.
```

> .. From the `components` section, set the `enabled` parameter to `true` for the `cnv-mtv-integrations-preview` component.

Author: ```suggestion
.. From the `components` specification, set the `enabled` parameter to `true` for the `cnv-mtv-integrations-preview` component.
```

> . Verify that the migration toolkit for virtualization is available on your cluster with the following command:

Reviewer: ```suggestion
. Verify that the migration toolkit for virtualization is available on your cluster. Run the following command:
```

Reviewer: The way this is written, it looks like it could be a "cluster with the following command..."

+
[source,bash]
----
oc get clustermanagementaddon mtv-operator  -o jsonpath='{.status.installProgressions[*].conditions[*].reason}
----
  
. Verify that the `kubevirt-hyperconverged` resource is available on your cluster with the following command:
  
+
[source,bash]
----
oc get clustermanagementaddon kubevirt-hyperconverged -o jsonpath='{.status.installProgressions[*].conditions[*].reason}
----
  
. Verify that the `kubevirt-hyperconverged-operator` is available on your cluster with the following command:

+
[source.bash]
----
oc get clustermanagementaddon kubevirt-hyperconverged-operator -o jsonpath='{.status.installProgressions[*].conditions[*].reason}
----

. To designate a cluster as a source or target for the virtual machine migration, apply the following label to your clusters: `acm/cnv-operator-install: "true"`.

. Enable cross-cluster migration from the {ocp-virt-short} console. Complete the following steps:
.. From the navigation menu, select *Virtualization* > *Overview*.
.. To access the preview features, click *Settings* > *Preview Features*.
.. Set `Enable Kubevirt cross cluster migration` to `true`.

> . *Optional:* Configure your network for the live migration within your `HyperConverged` resource by updating the `customizedVariables` section with the network key and value to your `cnv-hco-config` `AddOnDeploymentConfig` resource. Your `AddOnDeployment` resource might resemble the following sample:

Reviewer: same here with _section_

Reviewer: `customizedVariables` _specification_

Reviewer: (not italicized in the doc, just here for emphasis.

> . *Optional:* Configure your network for the live migration within your `HyperConverged` resource by updating the `customizedVariables` section with the network key and value to your `cnv-hco-config` `AddOnDeploymentConfig` resource. Your `AddOnDeployment` resource might resemble the following sample:

Author: ```suggestion
. *Optional:* Configure your network for the live migration within your `HyperConverged` resource by updating the `customizedVariables` specification. Add the network key and value to your `cnv-hco-config` `AddOnDeploymentConfig` resource. Your `AddOnDeployment` resource might resemble the following sample:
```

+
[source,yaml]
----
apiVersion: addon.open-cluster-management.io/v1alpha1
kind: AddOnDeploymentConfig
metadata:
  name: cnv-hco-config
  namespace: open-cluster-management
spec:
  agentInstallNamespace: openshift-cnv
  customizedVariables:
    - name: LIVE_NETWORK_KEY
      value: network
    - name: LIVE_NETWORK_VALUE
      value: {{ NetworkAttachmentDefinition name like lm-network }}
----

//is this sample between lines 82 and 97 necessary since there is no parameter for the network value?

> //// 

Reviewer: If we are intentionally commenting this out, we need to take it out and open an issue before merge to address it after merge if needed.

Author: @yiraeChristineKim is this part of the YAML important? I didn't see anything related to the network parameters. When I write "this part", I am referring to lines 82 through 97

Reviewer 2: I think the comment is not important. showing the yaml example + explaining NetworkAttachmentDefinition  is enough

[#migrate-vm]
= Migrating virtual machines between clusters (Technology Preview)

> Migrating virtual machines help you move resources during cluster upgrades, any time you need to clear a node for maintenance, or configuration changes. When you enable migration, the Kubevirt HyperConverged Operator is automatically installed on your managed clusters with the `acm/cnv-operator-install` label. The Migration Toolkit for Virtualization add-on Operator is installed on your hub cluster.

Reviewer 2: ```suggestion
Migrating virtual machines help you move resources during cluster upgrades, any time you need to clear a node for maintenance, or configuration changes. When you enable migration, the Kubevirt HyperConverged Operator is automatically installed on your managed clusters with the `acm/cnv-operator-install` label. The Migration Toolkit for Virtualization Operator is installed on your hub cluster.
```

.Prerequsites

- *Required access:* Cluster adminstrator
- You must have access to the `openshift-cnv` namespace.
- You have the latest version of {ocp-short} installed.

> //is it required for users to have the latest version of OCP installed to complete migration on the hub cluster? This helps determine what version of OCP i need to link to

Author: @yiraeChristineKim

Reviewer 2: Do we have a RBAC Doc for kubevirt.io:Admin?

Reviewer 2: 4.20 openshift

Reviewer 2: roles: kubevirt.io:admin, kubevirt.io-acm-hub:admin are enough for migration. This is related to ACM RBAC

Author: It is mentioned here: https://github.com/stolostron/rhacm-docs/blob/2.15_stage/secure_clusters/fine_grain_rbac_cli.adoc

Reminder that the RBAC still needs to be updated. Will you confirm if a ACM 2.15 user can complete this task with OCP 4.18?

Author: > roles: kubevirt.io:admin, kubevirt.io-acm-hub:admin are enough for migration. This is related to ACM RBAC

Are you saying that additional required roles are Kubevirt administrator?

Reviewer 2: CNV operator only supports 4.19 and higher

Author: Gotchu, thanks for confirming! That means OCP 4.19 is the minimum supported version. I will link to 4.19 OCP docs

To start migration for virtual machines complete the following steps:

. Enable `mtv-integrations-preview` component for the integration of {ocp-virt}. 
.. Run the following command to edit your `multiclusterhub` instance:

+
[source,bash]
----
oc edit MultiClusterHub multiclusterhub -n open-cluster-management
----

> .. From the `components` section, set the `enabled` parameter to `true` for the `mtv-integrations-preview` component.

Reviewer 2: ```suggestion
.. From the `components` section, set the `enabled` parameter to `true` for the `cnv-mtv-integrations-preview` component.
```
Sorry the name has been changed

. Verify that the Kubevirt `HyperConverged` Operator and Migration Toolkit for Virtualization add-on Operator are available on your cluster.

> //is there a way for the user to check this from the CLI? Not sure if we will doc UI steps

Author: ```suggestion
. Verify that the Migration Toolkit for Virtualization add-on Operator is available on your cluster with the following command:

+
[source,bash]
----
kubectl get clustermanagementaddon mtv-operator  -o jsonpath='{.status.installProgressions[*].conditions[*].reason}
----
  
. Verify that the Kubevirt `HyperConverged` resource is available on your cluster with the following command:
  
+
[source,bash]
----
kubectl get clustermanagementaddon kubevirt-hyperconverged -o jsonpath='{.status.installProgressions[*].conditions[*].reason}
----
  
. Verify that the Kubevirt `HyperConverged` Operator is available on your cluster with the following command:

+
[source.bash]
----
kubectl get clustermanagementaddon kubevirt-hyperconverged-operator -o jsonpath='{.status.installProgressions[*].conditions[*].reason}
----
```

> //is there a way for the user to check this from the CLI? Not sure if we will doc UI steps

Author: @yiraeChristineKim

Reviewer 2: ```
`kubectl get clustermanagementaddon mtv-operator \
  -o jsonpath='{.status.installProgressions[*].conditions[*].reason}'`

kubectl get clustermanagementaddon kubevirt-hyperconverged \
  -o jsonpath='{.status.installProgressions[*].conditions[*].reason}'

  kubectl get clustermanagementaddon kubevirt-hyperconverged-operator \
  -o jsonpath='{.status.installProgressions[*].conditions[*].reason}'
```

> . To designate a cluster as a source for the virtual machine migration, apply the following label to your clusters: `acm/cnv-operator-install: "true"`.

Reviewer 2: ```suggestion
. To designate a cluster as a source or target for the virtual machine migration, apply the following label to your clusters: `acm/cnv-operator-install: "true"`.
```

. Enable the Kubevirt cross-cluster migration from the {ocp-virt-short} console.
.. From the navigation menu, select *Virtualization* > *Overview*.
.. To access the preview features, click *Settings* > *Preview Features*.
.. Set _Enable Kubevirt cross cluster migration_ to `true`.

. _Optional:_ Configure your network for the live migration.

> .. To configure your network for managed clusters within your `HyperConverged` resource, update the `customizedVariables` section with the network key and value to your `cnv-config` `AddOnDeploymentConfig` resource. Your `AddOnDeployment` resource might resemble the following sample:

Reviewer 2: ```suggestion
.. To configure your network for managed clusters within your `HyperConverged` resource, update the `customizedVariables` section with the network key and value to your `cnv-hco-config` `AddOnDeploymentConfig` resource. Your `AddOnDeployment` resource might resemble the following sample:
``` 
sorry the name has been changed

+
[source,yaml]
----
apiVersion: addon.open-cluster-management.io/v1alpha1
kind: AddOnDeploymentConfig
metadata:

>   name: cnv-config

Reviewer 2: cnv-hco-config
```suggestion
  name: cnv-hco-config
```

  namespace: open-cluster-management
spec:
  agentInstallNamespace: openshift-cnv
  customizedVariables:
    - name: LIVE_NETWORK_KEY
      value: network
    - name: LIVE_NETWORK_VALUE
      value: {{ NetworkAttachmentDefinition name like lm-network }}
---
apiVersion: addon.open-cluster-management.io/v1alpha1
kind: ManagedClusterAddOn
metadata:
  name: kubevirt-hyperconverged-operator
  namespace: {{ managedcluster name }}
spec:
  installNamespace: openshift-cnv
  configs:
    - group: addon.open-cluster-management.io
      resource: addondeploymentconfigs

>       name: cnv-config

Reviewer 2: ```suggestion
      name: cnv-hco-config
```

[#migrate-vm]
= Migrating virtual machines between clusters (Technology Preview)

> Migrating virtual machines help you move resources during cluster upgrades, any time you need to clear a node for maintenance, or configuration changes. When you enable migration, the Kubevirt HyperConverged Operator is automatically installed on your managed clusters with the `acm/cnv-operator-install` label. The Migration Toolkit for Virtualization Operator is installed on your hub cluster.

Author: Need help confirming if I am mentioning the correct product names. took a look at our product list and saw inconsistencies here in the PR

```suggestion
Migrating virtual machines help you move resources during cluster upgrades, any time you need to clear a node for maintenance, or configuration changes. When you enable migration, the {ocp-virt} is automatically installed on your managed clusters with the `acm/cnv-operator-install` label. The migration toolkit for virtualization is also installed on your hub cluster.
```

Reviewer 2: "OpenShift Virtualization" this is official name

Reviewer 2: Migration Toolkit for Virtualization Operator
is correct

Author: Had a quick huddle w/Christine to confirm the names.

migration toolkit for virtualization is a separate operator

.Prerequsites

- *Required access:* Cluster administrator and Kubevirt adminstrator

> - You must have access to the `openshift-cnv` namespace.

Reviewer 2: mtv-integrations ns too. so this sentence is not enough so can we say like you need kubevirt.io:admin, kubevirt.io-acm-hub:admin roles more info you can see the RBAC doc somethiing like this

Reviewer 2: Are you saying that additional required roles are Kubevirt administrator? -> yes users need this roles." You must have access to the `openshift-cnv` namespace" this is not enough

Reviewer 2: also  Cluster administrator and Kubevirt adminstrator need kubevirt.io:admin role because of aggregate api

Reviewer 2: Summary: Remove this line and require the kubevirt.io:admin and kubevirt.io-acm-hub:admin roles for live migration. If users need full console access — including configurations, pods, metrics, StorageMaps, NetworkMaps, etc. — they must also have the kubevirt.io-acm-managed:admin or kubevirt.io-acm-managed:view o role on the managed cluster where the VMs reside. Users can simply add this in ACM User Management

> - You must have access to the `openshift-cnv` namespace.

Author: @yiraeChristineKim It seems like accessing the managed cluster is optional. Are you ok with me only mentioning what is required and not mention the "If" statement
```suggestion
- You have the `kubevirt.io:admin` and `kubevirt.io-acm-hub:admin` roles for live migration.
```

Reviewer 2: Good! We’ll also add "kubevirt.io-acm-managed:admin" to the managed cluster as both a target and a source cluster.

> - You have the latest version of {ocp-short} installed. See the link:https://docs.redhat.com/en/documentation/openshift_container_platform/4.19/html/overview/index[{ocp-short} Overview].

Author: ```suggestion
- You have the `kubevirt.io-acm-managed:admin` role as both a target and a source cluster.

- You have the latest version of {ocp-short} installed. See the link:https://docs.redhat.com/en/documentation/openshift_container_platform/4.19/html/overview/index[{ocp-short} Overview].
```

To start migration for virtual machines complete the following steps:

> . Enable `mtv-integrations-preview` component for the integration of {ocp-virt}. 

Reviewer 2: ```suggestion
. Enable `cnv-mtv-integrations-preview` component for the integration of {ocp-virt}. 
```

.. Run the following command to edit your `multiclusterhub` instance:

+
[source,bash]
----
oc edit MultiClusterHub multiclusterhub -n open-cluster-management
----

.. From the `components` section, set the `enabled` parameter to `true` for the `cnv-mtv-integrations-preview` component.

> . Verify that the Migration Toolkit for Virtualization add-on Operator is available on your cluster with the following command:

Author: ```suggestion
. Verify that the migration toolkit for virtualization is available on your cluster with the following command:
```

Reviewer 2: "Migration Toolkit for Virtualization Operator" this is official name

Author: Did a huddle and went through the official product name list

+
[source,bash]
----
kubectl get clustermanagementaddon mtv-operator  -o jsonpath='{.status.installProgressions[*].conditions[*].reason}
----
  
. Verify that the Kubevirt `HyperConverged` resource is available on your cluster with the following command:
  
+
[source,bash]
----
kubectl get clustermanagementaddon kubevirt-hyperconverged -o jsonpath='{.status.installProgressions[*].conditions[*].reason}
----
  

> . Verify that the Kubevirt `HyperConverged` Operator is available on your cluster with the following command:

Author: ```suggestion
. Verify that the HyperConverged Cluster Operator is available on your cluster with the following command:
```

Reviewer 2: "OpenShift Virtualization" This is official name

[#migrate-vm]
= Migrating virtual machines between clusters (Technology Preview)

Migrating virtual machines help you move resources during cluster upgrades, any time you need to clear a node for maintenance, or configuration changes. When you enable migration, the Kubevirt HyperConverged Operator is automatically installed on your managed clusters with the `acm/cnv-operator-install` label. The Migration Toolkit for Virtualization Operator is installed on your hub cluster.

.Prerequsites

- *Required access:* Cluster adminstrator
- You must have access to the `openshift-cnv` namespace.
- You have the latest version of {ocp-short} installed.

> //is it required for users to have the latest version of OCP installed to complete migration on the hub cluster? This helps determine what version of OCP i need to link to

Author: ```suggestion
- *Required access:* Cluster administrator and Kubevirt adminstrator
- You must have access to the `openshift-cnv` namespace.
- You have the latest version of {ocp-short} installed.
//is it required for users to have the latest version of OCP installed to complete migration on the hub cluster? This helps determine what version of OCP i need to link to
```

> //is it required for users to have the latest version of OCP installed to complete migration on the hub cluster? This helps determine what version of OCP i need to link to

Author: ```suggestion
- You have the latest version of {ocp-short} installed. See the link:https://docs.redhat.com/en/documentation/openshift_container_platform/4.19/html/overview/index[{ocp-short} Overview].
```

[#migrate-vm]
= Migrating virtual machines between clusters (Technology Preview)

Migrating virtual machines help you move resources during cluster upgrades, any time you need to clear a node for maintenance, or configuration changes. When you enable migration, the {ocp-virt} is automatically installed on your managed clusters with the `acm/cnv-operator-install` label. The migration toolkit for virtualization is also installed on your hub cluster.

.Prerequsites

- *Required access:* Cluster administrator
- You must have access to the `openshift-cnv` namespace.

> - You have the `kubevirt.io-acm-managed:admin` role as both a target and a source cluster.

Reviewer 2: ```suggestion
- You must have access to the `openshift-cnv` namespace.
- You have the `kubevirt.io:admin` role as both a target and a source cluster
- You have the `kubevirt.io-acm-hub:admin` for your hub cluster and `kubevirt.io-acm-managed:admin` for your target managed cluster. 
```

Author: Thanks @yiraeChristineKim, I updated your suggestion. I split the sentences because it was kind of long.

+
[source,bash]
----
oc edit MultiClusterHub multiclusterhub -n open-cluster-management
----

.. From the `components` section, set the `enabled` parameter to `true` for the `cnv-mtv-integrations-preview` component.

. Verify that the migration toolkit for virtualization is available on your cluster with the following command:

+
[source,bash]
----
kubectl get clustermanagementaddon mtv-operator  -o jsonpath='{.status.installProgressions[*].conditions[*].reason}
----
  
. Verify that the Kubevirt `HyperConverged` resource is available on your cluster with the following command:
  
+
[source,bash]
----
kubectl get clustermanagementaddon kubevirt-hyperconverged -o jsonpath='{.status.installProgressions[*].conditions[*].reason}
----
  
. Verify that the Kubevirt `HyperConverged` Operator is available on your cluster with the following command:

+
[source.bash]
----
kubectl get clustermanagementaddon kubevirt-hyperconverged-operator -o jsonpath='{.status.installProgressions[*].conditions[*].reason}
----

. To designate a cluster as a source or target for the virtual machine migration, apply the following label to your clusters: `acm/cnv-operator-install: "true"`.

> . Enable the Kubevirt cross-cluster migration from the {ocp-virt-short} console.

Author: ```suggestion
. Enable cross-cluster migration from the {ocp-virt-short} console.
```

[#migrate-vm]
= Migrating virtual machines between clusters (Technology Preview)

> Migrating virtual machines help you move resources during cluster upgrades, any time you need to clear a node for maintenance, or configuration changes. When you enable migration, the {ocp-virt} is automatically installed on your managed clusters with the `acm/cnv-operator-install` label. The migration toolkit for virtualization is also installed on your hub cluster.

Reviewer: ```suggestion
Migrating virtual machines helps you move resources during cluster upgrades, any time you need to clear a node for maintenance, or configuration changes. When you enable migration, the {ocp-virt} is automatically installed on your managed clusters with the `acm/cnv-operator-install` label. The migration toolkit for virtualization is also installed on your hub cluster.
```

Reviewer: That first line is off a bit, the phrases don't all match the subject or the commas are off, I cannot tell.

.Prerequsites

> - *Required access:* Cluster administrator

Reviewer: We can put these under the prereqs but we haven't been doing that.

Author: Got it, i updated it.

- You must have access to the `openshift-cnv` namespace.
- You have the `kubevirt.io:admin` role as both a target and a source cluster
- You have the `kubevirt.io-acm-hub:admin` for your hub cluster and `kubevirt.io-acm-managed:admin` for your target managed cluster. 

- You have the latest version of {ocp-short} installed. See the link:https://docs.redhat.com/en/documentation/openshift_container_platform/4.19/html/overview/index[{ocp-short} Overview].

To start migration for virtual machines complete the following steps:

. Enable `cnv-mtv-integrations-preview` component for the integration of {ocp-virt}. 
.. Run the following command to edit your `multiclusterhub` instance:

+
[source,bash]
----
oc edit MultiClusterHub multiclusterhub -n open-cluster-management
----

> .. From the `components` section, set the `enabled` parameter to `true` for the `cnv-mtv-integrations-preview` component.

Reviewer: ```suggestion
.. From the _components_ section, set the `enabled` parameter to `true` for the `cnv-mtv-integrations-preview` component.
```

Reviewer: Does that get capitalized?

Author: @yiraeChristineKim should components be capitalized?

Reviewer 2: No. in manifest, it is "components"

Reviewer: ok, so this is describing YAML, I see. 

From the `components` (section?); we normally say _setting_ or _specification_ for something like that. Do either of those work?

. Verify that the migration toolkit for virtualization is available on your cluster with the following command:

+
[source,bash]
----
kubectl get clustermanagementaddon mtv-operator  -o jsonpath='{.status.installProgressions[*].conditions[*].reason}
----
  
. Verify that the Kubevirt `HyperConverged` resource is available on your cluster with the following command:
  
+
[source,bash]

> ----

Reviewer: Why are we using kubectl? A long time ago, we changed almost everything to `oc` unless ocp is not required, very few places do we use kubectl.

Author: Thanks for the reminder! From my understanding, OCP is required. @yiraeChristineKim for this task, will the commands function as expected if we use "oc" instead of "kubectl". Heads up, I did already update the commands but will revert if necessary

Reviewer 2: we can use `oc`

kubectl get clustermanagementaddon kubevirt-hyperconverged -o jsonpath='{.status.installProgressions[*].conditions[*].reason}
----
  

> . Verify that the Kubevirt `HyperConverged` Operator is available on your cluster with the following command:

Reviewer: Is this the name?  Kubevirt `HyperConverged` Operator

Reviewer 2: this is addon name

Reviewer: The way this is written:  Kubevirt `HyperConverged` Operator 

It doesn't make sense because part of the name is coded.

What is the actual approved name, and then we should not code part of it. What does the naming guidance say about this, because I have not seen Kubevirt used outside of the role. Just looking for the official name here.

Author: I monospaced it because it is monospaced in the OCP, but after taking another look I see they only monospaced when referring to the custom resource. removing the monospace around "HyperConverged"

Jk! I actually kept the monospace, but ensured I matched the name from the command.

Reviewer: It does not make sense to have that in the middle of the operator name. What is the actual operator name? OCP is not always correct, so best to check for the name. We would not do something like:

`multicluster-engine` Operator

We code it `multicluster-engine` or we name it **multicluster engine operator** (only lower case bc that is how it was in naming guidance.

+
[source.bash]
----
kubectl get clustermanagementaddon kubevirt-hyperconverged-operator -o jsonpath='{.status.installProgressions[*].conditions[*].reason}
----

. To designate a cluster as a source or target for the virtual machine migration, apply the following label to your clusters: `acm/cnv-operator-install: "true"`.

. Enable cross-cluster migration from the {ocp-virt-short} console.
.. From the navigation menu, select *Virtualization* > *Overview*.
.. To access the preview features, click *Settings* > *Preview Features*.

> .. Set _Enable Kubevirt cross cluster migration_ to `true`.

Reviewer: ```suggestion
.. Set `Enable Kubevirt cross cluster migration` to `true`.
```

Author: "Enable Kubevirt cross cluster migration" is in the OCP virt console. That's why I italicized it. I'll change it tho

Reviewer: Ok, if it is a setting, we typically code it. We italicize it if is a window or a field, so I cannot say without seeing it what that should be. They may not be using the right code in the UI.

Just like YAML, we would tell them to set `a-named-value` to `true`.

Reviewer: it may make sense to violate that rule here.

> . _Optional:_ Configure your network for the live migration.

Reviewer: ```suggestion
. *Optional:* Configure your network for the live migration.
```

.. To configure your network for managed clusters within your `HyperConverged` resource, update the `customizedVariables` section with the network key and value to your `cnv-hco-config` `AddOnDeploymentConfig` resource. Your `AddOnDeployment` resource might resemble the following sample:

+
[source,yaml]
----
apiVersion: addon.open-cluster-management.io/v1alpha1
kind: AddOnDeploymentConfig
metadata:
  name: cnv-hco-config
  namespace: open-cluster-management
spec:
  agentInstallNamespace: openshift-cnv
  customizedVariables:
    - name: LIVE_NETWORK_KEY
      value: network
    - name: LIVE_NETWORK_VALUE
      value: {{ NetworkAttachmentDefinition name like lm-network }}

> ---

Reviewer: There are two resources here, but we don't explain that, and it really should be broken out.
